src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -470,7 +495,7 @@ public static string Concat(params string?[] values)
                         break;
```

Comment:
throw OutOfMemoryException can use the new helper with the better error message.